joomla/joomla-cms
# fix mime checkes

Pull Request for Issue #16086

### Summary of Changes

If you have installed 3.7.1 and try to upload using com_media a file which is not a image you got an error that you can't upload it.

I have now:
- [X] Improved the checks
- [X] show the failing mime to the use
- [X] make sure that images are checked with the image functions and other files with the other checks
- [X] implemented a fallback for images
- [X] fixed a longstanding issuse with finfo_open (FILEINFO_MIME vs. FILEINFO_MIME_TYPE)

### Testing Instructions

- install 3.7.1
- try to upload a PDF file.
- it fails
- try to upload a jpg file
- it works
- apply this patch
- try the PDF and the image again
- it works now

Now go to the options and don't allow uploading jpg images and PDF files (by removing the jpg & PDF mime from the allowed mime types

- try to upload a jpg again
- it should not work but now show the failing mime type. (image/jpg)
- do the same for a PDF
- it should also show the failing mime.

### Expected result

If allowed the upload should work if not it should show the mime we detected to the user.

### Actual result

non image files can't be uploaded using the media manger

### Documentation Changes Required

none.
